### PR TITLE
fix bug in DefaultFileAbstractLayer for copy.

### PR DESCRIPTION
**Pull request by vwxyzh**
@chenkennt @ansyral @hellosnow @vicancy @qinezh @superyyrrzz

### CODE REVIEW

**Comment by hellosnow** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Plugins/RootedFileAbstractLayer.cs [hunk 1]
```diff
@@ -45,7 +45,8 @@ public void Copy(string sourceFileName, string destFileName)
             if (Path.IsPathRooted(sourceFileName) || Path.IsPathRooted(destFileName))
             {
                 Directory.CreateDirectory(Path.GetDirectoryName(destFileName));
-                File.Copy(sourceFileName, destFileName);
+                File.Copy(sourceFileName, destFileName, true);
+                File.SetAttributes(destFileName, FileAttributes.Normal);
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Plugins/RootedFileAbstractLayer.cs]: why SetAttributes is needed?

Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Plugins/RootedFileAbstractLayer.cs]: copy a read only file, then copy it again.

Comment (vicancy) [inline src/Microsoft.DocAsCode.Plugins/RootedFileAbstractLayer.cs]: Extract to a common method?

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by qinezh** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Plugins/DefaultFileAbstractLayer.cs [hunk 2]
```diff
@@ -47,7 +47,7 @@ public void Copy(string sourceFileName, string destFileName)
             var source = GetPhysicalPath(sourceFileName);
             var dest = GetOutputPhysicalPath(destFileName);
             Directory.CreateDirectory(Path.GetDirectoryName(dest));
-            File.Copy(source, dest);
+            File.Copy(source, dest, true);
```
Comment (ansyral) [inline src/Microsoft.DocAsCode.Plugins/DefaultFileAbstractLayer.cs]: do we need to add a parameter Overwrite in Copy method?

Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Plugins/DefaultFileAbstractLayer.cs]: No for now.

**Pull request merged**